### PR TITLE
Support for Datatype String()

### PR DESCRIPTION
**Pull request by WestleyK**
This PR adds a `String()`  function for `Datatype`s.

 - Closes: #67

### CODE REVIEW

File: pkg/sif/fmt.go [hunk 1]
```diff
@@ -11,6 +11,28 @@ import (
...
+	case DataSignature:
+		return "Signature"
+	case DataGenericJSON:
+		return "JSON.Generic"
+	case DataGeneric:
+		return "Generic/Raw"
+	case DataCryptoMessage:
+		return "Cryptographic Message"
+	}
+	return "Unknown data-type"
```
Comment (sashayakovtseva) [inline pkg/sif/fmt.go]:
```suggestion
	return "Unknown"
```

File: pkg/sif/fmt.go [hunk 2]
```diff
@@ -184,16 +184,16 @@ func (fimg *FileImage) FmtDescrList() string {
 				f, _ := v.GetFsType()
 				p, _ := v.GetPartType()
 				a, _ := v.GetArch()
-				s += fmt.Sprintf("|%s (%s/%s/%s)\n", datatypeStr(v.Datatype), fstypeStr(f), parttypeStr(p), GetGoArch(trimZeroBytes(a[:])))
+				s += fmt.Sprintf("|%s (%s/%s/%s)\n", v.Datatype.String(), fstypeStr(f), parttypeStr(p), GetGoArch(trimZeroBytes(a[:])))
```
Comment (sashayakovtseva) [inline pkg/sif/fmt.go]: no need to call string() explicitly

Comment (mem) [reply pkg/sif/fmt.go]:
```suggestion
				s += fmt.Sprintf("|%s (%s/%s/%s)\n", v.Datatype, fstypeStr(f), parttypeStr(p), GetGoArch(trimZeroBytes(a[:])))
```

File: pkg/sif/fmt.go [hunk 3]
```diff
@@ -184,16 +184,16 @@ func (fimg *FileImage) FmtDescrList() string {
...
 			case DataSignature:
 				h, _ := v.GetHashType()
-				s += fmt.Sprintf("|%s (%s)\n", datatypeStr(v.Datatype), hashtypeStr(h))
+				s += fmt.Sprintf("|%s (%s)\n", v.Datatype.String(), hashtypeStr(h))
```
Comment (mem) [inline pkg/sif/fmt.go]:
```suggestion
				s += fmt.Sprintf("|%s (%s)\n", v.Datatype, hashtypeStr(h))
```

File: pkg/sif/fmt.go [hunk 4]
```diff
@@ -184,16 +184,16 @@ func (fimg *FileImage) FmtDescrList() string {
...
 			case DataCryptoMessage:
 				f, _ := v.GetFormatType()
 				m, _ := v.GetMessageType()
-				s += fmt.Sprintf("|%s (%s/%s)\n", datatypeStr(v.Datatype), formattypeStr(f), messagetypeStr(m))
+				s += fmt.Sprintf("|%s (%s/%s)\n", v.Datatype.String(), formattypeStr(f), messagetypeStr(m))
```
Comment (mem) [inline pkg/sif/fmt.go]:
```suggestion
				s += fmt.Sprintf("|%s (%s/%s)\n", v.Datatype, formattypeStr(f), messagetypeStr(m))
```

File: pkg/sif/fmt.go [hunk 5]
```diff
@@ -210,7 +210,7 @@ func (fimg *FileImage) FmtDescrInfo(id uint32) string {
 			continue
 		} else if v.ID == id {
 			s = fmt.Sprintln("Descr slot#:", i)
-			s += fmt.Sprintln("  Datatype: ", datatypeStr(v.Datatype))
+			s += fmt.Sprintln("  Datatype: ", v.Datatype.String())
```
Comment (mem) [inline pkg/sif/fmt.go]:
```suggestion
			s += fmt.Sprintln("  Datatype: ", v.Datatype)
```

**Pull request merged**